pyth_observer/event.py

```diff
@@ -94,3 +99,44 @@ async def send(self):
 
         level = cast(LogEventLevel, os.environ.get("LOG_EVENT_LEVEL", "INFO"))
         logger.log(level, text.replace("\n", ". "))
+
+
+class TelegramEvent(Event):
+    def __init__(self, check: Check, context: Context):
+        self.check = check
+        self.context = context
+
+    async def send(self):
+        if self.check.__class__.__bases__ == (PublisherCheck,):
+            text = self.check.error_message()
+            publisher_key = self.check.state().public_key.key
+            publisher = self.context["publishers"].get(publisher_key, None)
+            # Ensure publisher is not None and has contact_info before accessing telegram_chat_id
+            chat_id = (
+                publisher.contact_info.telegram_chat_id
+                if publisher is not None and publisher.contact_info is not None
+                else None
+            )
+
+            if chat_id is None:
+                logger.warning(
+                    f"Telegram chat ID not found for publisher key {publisher_key}"
+                )
+                return
+
+            telegram_api_url = f"https://api.telegram.org/bot{os.environ['TELEGRAM_BOT_TOKEN']}/sendMessage"
```

Comment:
Can you load this environment variable in the constructor? So we can catch it earlier if it's not passed.